activelearning/dataset/ocp.py

```diff
@@ -41,31 +43,41 @@ def get_item_from_index(self, index):
         # return scatter(hidden_states, state_idcs, dim=0, reduce="add"), torch.Tensor(
         #     [datapoint.y_relaxed]
         # )
-        return hidden_states.mean(0).unsqueeze(0), torch.Tensor([datapoint.y_relaxed])
+        return (
+            hidden_states.mean(0).unsqueeze(0),
+            torch.Tensor([datapoint.y_relaxed]),
+            torch.Tensor([datapoint.idx_in_dataset]),
+        )
 
     def __getitem__(self, key):
         if isinstance(key, int):
-            x, y = self.get_item_from_index(key)
+            x, y, idcs_in_dataset = self.get_item_from_index(key)
         elif isinstance(key, slice):
             start, stop, step = key.indices(len(self))
             x = torch.Tensor([])
             y = torch.Tensor([])
+            idcs_in_dataset = torch.Tensor([])
             for i in range(start, stop, step):
```

Comment:
I would not `concat` sequentially in a `for` loop, but either:

* store in a list and `concat` afterwards
* pre-allocate a large tensor and fill it (~`large[a:b, ...] = somehting`)

Reply:
does this have an effect on performance? or just bad practice?